haigha/classes/basic_class.py

```diff
@@ -189,10 +212,23 @@ def return_msg(self, reply_code, reply_text, exchange, routing_key):
 
         self.send_frame(MethodFrame(self.channel_id, 60, 50, args))
 
-    def _recv_return(self, _method_frame):
-        # This seems like the right place to callback that the operation has
-        # completed.
-        pass
+    def _recv_return(self, method_frame):
+        '''
+        Handle basic.return method. If we have a complete message, will call the
+        user's return listener callabck (if any). If there are not enough
+        frames, will re-queue current frames and raise a FrameUnderflow
+
+        NOTE: if the channel was in confirmation mode when the message was
+        published, then this will still be followed by basic.ack later
+        '''
+        msg = self._read_returned_msg(method_frame)
+
+        if self._return_listener is not None:
```

Comment:
I'm fine with this, but perhaps the following would be better. I've been trying to switch to this usage in new code I write.

```python
if callable(self._return_listener):
```